modules/artifact/src/main/java/org/mule/runtime/module/artifact/api/classloader/ModuleLayerInformationSupplier.java

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright 2023 Salesforce, Inc. All rights reserved.
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.artifact.api.classloader;
+
+public interface ModuleLayerInformationSupplier {
+
+  String retrieveInformation();
```

Comment:
`graphRepresentation`?

Reply:
I didn't want it to tie it to be a graph representation, in case we just wanted any other representation (e.g. just the  text of which modules are involved in any module layer, or anything else we might think of). I will change it to `retrieveRepresentation` though